common/models/model-property.js

```diff
@@ -0,0 +1,60 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback-workspace
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+'use strict';
+
+var path = require('path');
+var fs = require('fs-extra');
+
+module.exports = function(ModelProperty) {
+  /**
+   * Represents a Property of a LoopBack `Model`.
+   *
+   * @class ModelProperty
+   * @inherits WorkspaceEntity
+   */
```

Comment:
We should also name struture the file differently -- i'm thinking like:

```
module.exports = ModelProperty;


function ModelProperty(..) {
}

ModelProperty.getAvailableTypes = function(...) {
}

...

and so forth to prevent the nesting.